### DISCUSSION TITLE
Automatic quick sync to download new purchases

### DISCUSSION THREAD

**Discussion by brainnoms** [openaudible/openaudible#1081] [Ideas]
So, if I'm in the audible client and I buy something new, it seems to automatically realize that I've purchased a new book.  It doesn't look like its doing so from an active refresh of the library.   I'd presume it's doing something like an RSS feed or an XMPP channel subscription or something that lets it realize a book was added to the library.  Certainly something that avoids actively polling every so many minutes.

I'd love to see similar behavior available in OpenAudible (especially since I run it in a container 24/7).  It could recognize there's a new book and trigger the equivalent of a quick sync and download the new book.  I just don't know how the app is getting notified so I'm stuck on any concrete decision as to how its working under the covers.

**Comment by openaudible** [comment]
There is a hidden feature to do this.. Edit the preferences.json file and set autoRefreshMinutes to some value-- 120 to automatically check every 2 hours. Save and restart openaudible...

**Reply by SpectreGadgetReborn** [reply]
Is there anything else needed for this to work?  I've updated to the latest beta.  When I tried updating it 60 minutes, when I load OA, it sets it to 0.  I change it to 120 or more, and it stays, but never auto-refreshes for me.  Running on Debian 12.8

**Reply by openaudible** [reply]
Make sure you edit the settings when the app isn't running as it can overwrite the settings. 

When the app closes double check the settings are still set. 

Check the log message for anything unusual.

**Reply by SpectreGadgetReborn** [reply]
Thanks.  I've done all that, but just did it again with a 60 minute refresh, so I'll look in the log after if it doesn't do it.

**Comment by openaudible** [comment]
I looked at the logic. Only real way to tell is you'll see a message like:
`Scanning 1 accoun(s). ` (Fixed the typo for the next release.)

Also make sure your Library Status says "Connected: Yes"

You can try it with it set to 5 minutes to make it easier to wait for that first refresh. 

But the big thing is to make sure the settings stay set in the settings.json file.

**Reply by SpectreGadgetReborn** [reply]
Thanks again. It seems to be working, although I didn't really do anything different that I know of.  I did make sure the value held after loading and then added a plus item to the library to see it happen.

**Comment by openaudible** [comment]
That was probably it.. We added a little more logging for the next release. Cheers.
